Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -503,34 +516,50 @@ protected class ClassWriter : IClassWriter {
       Contract.Requires(wr != null);
 
       return type.NormalizeExpandKeepConstraints() switch {
+        var x when x.IsBuiltinArrowType => $"{DafnyDefaults}.pointer",
         // unresolved proxy; just treat as bool, since no particular type information is apparently needed for this type
         BoolType or TypeProxy => $"{DafnyDefaults}.bool",
```

Comment:
Would we be able to catch errors earlier if we returned a dummy object in the `TypeProxy` case?

Reply:
This more explicit workaround is unique to the js compiler, for now. The others must catch it more implicitly. changing `NormalizeExpandKeepConstraints` may not be the worst idea